src/ModuleLoader.ts

```diff
@@ -387,7 +404,35 @@ export class ModuleLoader {
 			return {
 				external: true,
 				id: source,
-				moduleSideEffects: this.hasModuleSideEffects(source, true)
+				moduleSideEffects: this.hasModuleSideEffects(source, true),
+				syntheticNamedExports: false
+			};
+		}
+		return resolvedId;
+	}
+
+	private handleResolveId(
+		resolvedId: ResolvedId | null,
+		source: string,
+		importer: string
+	): ResolvedId {
+		return this.handleSyntheticExports(
+			this.handleMissingImports(resolvedId, source, importer),
+			source,
+			importer
+		);
+	}
+
+	private handleSyntheticExports(
```

Comment:
Those two methods are rather short, can they be merged or is the latter also used separately?